kola/harness.go

```diff
@@ -369,7 +369,7 @@ func runTest(h *harness.H, t *register.Test, pltfrm string) {
 
 // architecture returns the machine architecture of the given platform.
 func architecture(pltfrm string) string {
-	nativeArch := "amd64"
+	nativeArch := system.PortageArch()
```

Comment:
If kola is running on arm64 but `pltfrm` is e.g. `aws`, `architecture` should still return `amd64`.  It looks like the right place to fix the default selection is in `sdk.DefaultBoard()`.  That will fix `QEMUOptions.Board` via `defaultTargetBoard` in `cmd/kola/options.go`.